.github/workflows/changesets.yml

```diff
@@ -0,0 +1,59 @@
+name: Changesets
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```

Comment:
Maybe run as `pull_request_target`? Otherwise, `${{ secrets.BOT_PAT }}` cannot be resolved in `pull_request` (fork repo) and always fails.

Or generally, I suspect why we need to trigger changeset generation for pull requests.